functests/4_latency/latency.go

```
@@ -229,62 +235,74 @@ var _ = Describe("[performance] Latency Test", func() {
+		if val, err = strconv.Atoi(latencyTestDelayEnv); err != nil {
+			err = fmt.Errorf("the environment variable LATENCY_TEST_DELAY has incorrect value %q", latencyTestDelayEnv)
 		}
+
 	}
+	return val, err
 }
 
 //read LATENCY_TEST_CPUS environment var and parse it if set
-func getLatencyTestCpus() {
+func getLatencyTestCpus() (parsedVal int, errMsg error) {
+	var err error
+	var val int
 	latencyTestCpusEnv := os.Getenv("LATENCY_TEST_CPUS")
 	if latencyTestCpusEnv != "" {
-		var err error
-		if latencyTestCpus, err = strconv.Atoi(latencyTestCpusEnv); err != nil {
-			klog.Fatalf("the environment variable LATENCY_TEST_CPUS has incorrect value %q", latencyTestCpusEnv)
+		if val, err = strconv.Atoi(latencyTestCpusEnv); err != nil {
+			err = fmt.Errorf("the environment variable LATENCY_TEST_CPUS has incorrect value %q", latencyTestCpusEnv)
 		}
 	}
+	return val, err
 }
 
-//read MAXIMUM_LATENCY environment var and parse it if set
-func getMaxLatency() {
+// getMaximumLatency should look for one of the following environment variables:
+// OSLAT_MAXIMUM_LATENCY: the expected maximum latency for all buckets in us
+// CYCLICTEST_MAXIMUM_LATENCY: the expected maximum latency for all buckets in us
+// HWLATDETECT_MAXIMUM_LATENCY: the expected maximum latency for all buckets in us
+// MAXIMUM_LATENCY: unified expected maximum latency for all tests
+func getMaximumLatency(testName string) (parsedValue int, errMsg error) {
+	var err error
+	var val int
 	unifiedMaxLatencyEnv := os.Getenv("MAXIMUM_LATENCY")
 	if unifiedMaxLatencyEnv != "" {
 		var err error
-		if maximumLatency, err = strconv.Atoi(unifiedMaxLatencyEnv); err != nil {
-			klog.Fatalf("the environment variable MAXIMUM_LATENCY has incorrect value %q", unifiedMaxLatencyEnv)
+		if val, err = strconv.Atoi(unifiedMaxLatencyEnv); err != nil {
+			err = fmt.Errorf("the environment variable MAXIMUM_LATENCY has incorrect value %q", unifiedMaxLatencyEnv)
```

You should return the error here, otherwise, you might override it later, when you're trying to parse the other env variables.

---

correct. In addition you should wrap errors here: https://blog.golang.org/go1.13-errors
TL;DR: use something like
```
err = fmt.Errorf("the environment variable MAXIMUM_LATENCY has incorrect value %q: %w", unifiedMaxLatencyEnv, err)
```

---

fixed, thanks.

---

@fromanirh please keep me honest here, but yet we should have a ```return``` statement here. Storing the error in ```err``` won't keep it from being override later on in the function.

---

it won't indeed

---

After discussing the matter with Talor, I added a return statement on the first error check.Thanks.